### PR TITLE
Core/Conditions: Add CONDITION_SOURCE_TYPE_QUEST_AVAILABLE to condition manager.

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed**: Add the ability for quest availability to be determined in condition table if player has completed the required quest. This is my first pull request and is a small partial fix to minimize the complaints.

**Target branch(es)**: 335/6x

**Issues addressed**: Inability of core to properly make a quest available only after two previous quest are completed. 

**Tests performed**: It builds.

**Known issues and TODO list**:
- [ ] Need Gossip system to use this condition to determine if quest is made available.

### CODE REVIEW

File: src/server/game/Conditions/ConditionMgr.cpp [hunk 1]
```diff
@@ -58,6 +58,7 @@ char const* const ConditionMgr::StaticSourceTypeData[CONDITION_SOURCE_TYPE_MAX]
     "Spell Proc",
     "Terrain Swap",
     "Phase"
```
Comment (Nyeriah) [inline src/server/game/Conditions/ConditionMgr.cpp]: Missing comma?

**Comment by DDuarte** [comment]
> Inability of core to properly make a quest available only after two previous quest are completed.

See https://trinitycore.atlassian.net/wiki/display/tc/quest_template **Chain of quests with multiple start quests.**. If that doesn't work anymore then it should be fixed instead of adding yet another way to do the same thing.

**Comment by malcrom** [comment]
I complained about this over a year ago while spawning Felwood, nothing was done. Anyway I am looking at CONDITION_SOURCE_TYPE_QUEST_SHOW_MARK to see it it can prevent quest from being added to gossip rather than checking gossip to hide the quest mark.

**Comment by DDuarte** [comment]
Can you paste code showing that the method described in the quest_template wiki does not work?

**Comment by malcrom** [comment]
Can't test with current 6.X core. Tkrokli tried last night.

**Comment by DDuarte** [comment]
It's the same "system" on 3.3.5

**Pull request closed without merging**